envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/routes/EnvoyEgressRoutesFactory.kt

```diff
@@ -177,3 +243,78 @@ class EnvoyEgressRoutesFactory(
         return routeAction
     }
 }
+
+class RequestPolicyMapper private constructor() {
+    companion object {
+        fun mapToEnvoyRetryPolicyBuilder(retryPolicy: EnvoyControlRetryPolicy?): RetryPolicy? {
+            retryPolicy?.let { policy ->
+                val retryPolicyBuilder = RetryPolicy.newBuilder()
+
+                policy.retryOn?.let { retryPolicyBuilder.setRetryOn(it) }
+                policy.hostSelectionRetryMaxAttempts?.let {
+                    retryPolicyBuilder.setHostSelectionRetryMaxAttempts(it)
+                }
+                policy.numberRetries?.let { retryPolicyBuilder.setNumRetries(UInt32Value.of(it)) }
+                policy.retryHostPredicate?.let {
+                    buildRetryHostPredicate(it, retryPolicyBuilder)
+                }
+                policy.perTryTimeoutMs?.let { retryPolicyBuilder.setPerTryTimeout(Durations.fromMillis(it)) }
+                policy.retryBackOff?.let {
+                    buildRetryBackOff(it, retryPolicyBuilder)
+                }
+                policy.retryableStatusCodes?.let {
+                    buildRetryableStatusCodes(it, retryPolicyBuilder)
+                }
+                policy.retryableHeaders?.let {
+                    buildRetryableHeaders(it, retryPolicyBuilder)
+                }
+
+                return retryPolicyBuilder.build()
+            }
+            return null
+        }
+
+        private fun buildRetryableHeaders(
+            it: List<String>,
+            retryPolicyBuilder: RetryPolicy.Builder
+        ) {
+            it.forEach { header ->
+                retryPolicyBuilder.addRetriableHeaders(
+                    HeaderMatcher.newBuilder().setName(header)
+                )
+            }
+        }
+
+        private fun buildRetryableStatusCodes(
+            it: List<Int>,
+            retryPolicyBuilder: RetryPolicy.Builder
+        ) {
+            it.forEach { statusCode -> retryPolicyBuilder.addRetriableStatusCodes(statusCode) }
```

Comment:
Why you not use `RetryPolicy.Builder#addAllRetriableStatusCodes` it make it simpler and then this method is useless